framework/source/class/qx/Bootstrap.js

```diff
@@ -401,6 +413,99 @@ qx.Bootstrap.define("qx.Bootstrap",
       construct.self = clazz.constructor = proto.constructor = clazz;
     },
 
+    /** Private list of classes whcih have a defer method that needs to be executed */
+    __pendingDefers: [],
+
+    /**
+     * Adds a callback for a class so that it's defer method can be called, either after all classes
+     * are loaded or when absolutely necessary because of load-time requirements of other classes.
+     *
+     * @param clazz
+     * @param cb
+     */
+    addPendingDefer: function(clazz, cb) {
+      if (qx.$$loader.delayDefer) {
+        this.__pendingDefers.push(clazz);
+        clazz.$$pendingDefer = cb;
+      } else {
+        cb.call(clazz);
+      }
+    },
+
+    /**
+     * Executes the defer methods for classes which are required by the dependency information in
+     * dbClassInfo (which is a map in the format generated by qxcompiler).  Defer methods are of course
+     * only eecuted once but they are always put off until absolutely necessary to avoid potential
+     * side effects and recursive and/or difficult to resolve dependencies.
+     *
+     * @param dbClassInfo
+     */
+    executePendingDefers: function(dbClassInfo) {
+      var t = this;
+      if (!dbClassInfo) {
+        var pendingDefers = this.__pendingDefers;
+        this.__pendingDefers = [];
+        pendingDefers.forEach(execute);
+        return;
+      }
+
+      function executeForDbClassInfo(dbClassInfo) {
+        if (dbClassInfo.environment) {
```

Comment:
I am rather fond of using 

`var executeForDbClassInfo = function(dbClassInfo) { `

for localized code references as it makes the locality of the definition more obvious.

what do you think?